pkg/codegen/python/gen.go

```diff
@@ -2119,6 +2101,46 @@ func (mod *modContext) genType(w io.Writer, name, comment string, properties []*
 		printComment(w, comment, "    ")
 	}
 
+	// To help users migrate to using the properly snake_cased property getters, emit warnings when camelCase keys are
+	// accessed. We emit this at the top of the class in case we have a `get` property that will be redefined later.
+	if !input && !functionType {
+		var needsCaseWarning bool
+		for _, prop := range props {
+			pname := PyName(prop.Name)
+			if pname != prop.Name {
+				needsCaseWarning = true
+				break
+			}
+		}
+		if needsCaseWarning {
+			fmt.Fprintf(w, "    @staticmethod\n")
+			fmt.Fprintf(w, "    def __key_warning(key: str):\n")
+			fmt.Fprintf(w, "        suggest = None\n")
+			prefix := "if"
+			for _, prop := range props {
+				pname := PyName(prop.Name)
+				if pname == prop.Name {
+					continue
+				}
+				fmt.Fprintf(w, "        %s key == \"%s\":\n", prefix, prop.Name)
+				fmt.Fprintf(w, "            suggest = \"%s\"\n", pname)
```

Comment:
Depending on whether we want to stick to single or double quotes, you might take the suggestion or change it to `'%s'`

```suggestion
				fmt.Fprintf(w, "        %s key == %q:\n", prefix, prop.Name)
				fmt.Fprintf(w, "            suggest = %q\n", pname)
```